documentation/design-docs/diagnostics-client-library.md

```diff
@@ -0,0 +1,327 @@
+                                    break;
+                                fs.Write(buffer, 0, nBytesRead);
+                            }
+                            DiagnosticsClient.StopTracing(traceSession);
+                            DiagnosticsClient.StopTracing(counterSession);
+                            shouldExit.Set();
+                        });
+                    }
+                }
+            }
+        }
+    }
+    try
+    {
+        source.Process();
+        shouldExit.WaitOne();
+    }
+    catch (Exception e) { } 
+}
+```
+
+## API Descriptions
+
+At a high level, the DiagnosticsClient class provides static methods that the user may call to invoke diagnostics IPC commands (i.e. start an EventPipe session, request a core dump, etc.) The library also provides several classes that may be helpful for invoking these commands. These commands are described in more detail in the diagnostics IPC protocol documentation available here: https://github.com/dotnet/diagnostics/blob/master/documentation/design-docs/ipc-protocol.md#commands. 
+
+### ProcessLocator
+This is a helper class that finds processes to attach to.
+```cs
+namespace Microsoft.Diagnostics.NETCore.Client
+{
+    /// <summary>
+    /// A utility class that contain various helper methods to interact with the diagnostics IPC.
+    /// </summary>
+    public class ProcessLocator
+    {
+        /// <summary>
+        /// Get all the active processes that can be attached to.
+        /// </summary>
+        /// <returns>
+        /// IEnumerable of all the active process IDs.
+        /// </returns>
+        public static IEnumerable<int> GetPublishedProcesses();
+    }
+}
+```
+
+
+### DumpType (enum)
+This is an enum for the dump type
+
+```cs
+namespace Microsoft.Diagnostics.NETCore.Client
+{
+    public enum DumpType
+    {
+        Normal = 1,
+        WithHeap = 2,
+        Triage = 3,
+        Full = 4
+    }
+}
+```
+
+
+### DiagnosticsClient
+This is a top-level class with static methods to send various diagnostics command to the runtime.
+```cs
+namespace Microsoft.Diagnostics.NETCore.Client
+{
+    public class DiagnosticsClient
+    {
+        /// <summary>
+        /// Start tracing the application via CollectTracing2 command.
+        /// </summary>
+        /// <param name="processId">Target process' ID</param>
+        /// <param name="providers">An IEnumerable containing the list of Providers to turn on.</param>
+        /// <param name="circularBufferMB">The size of the runtime's buffer for collecting events in MB</param>
+        /// <param name="requestRundown">If true, request rundown events from the runtime</param>
+        /// <returns>
+        /// An EventPipeSession object representing the EventPipe session that just started.
+        /// </returns> 
+        public static EventPipeSession StartTracing(int processId, IEnumerable<EventPipeProvider> providers, int circularBufferMB=256, bool requestRundown=true)
```

Comment:
I'd suggest making these instance methods and use the constructor to handle connection information (processId). There is a decent chance that in the future we'd need richer connection information such as pipe name and it would be unfortunate if we had to create overloads of every API to support the different possible addressing schemes.